internal/sysclient/sysclient.go

```diff
@@ -87,48 +89,162 @@ func (s *SysClient) JszPing(opts server.JszEventOptions, fopts ...FetchOpt) ([]J
 	return srvJsz, nil
 }
 
+func (s *SysClient) FetchJszPaged(baseOpts server.JszEventOptions, limit int, timeout time.Duration, fn func([]*server.AccountDetail)) error {
+	offsets := map[string]int{}
+	fopts := []FetchOpt{fetchTimeout(timeout)}
+	baseOpts.Limit = limit
+
+	// Initial request
+	initialPages, err := s.JszPing(baseOpts, fopts...)
+	if err != nil {
+		return err
+	}
+
+	for _, page := range initialPages {
+		name := page.Server.Name
+		fn(page.JSInfo.AccountDetails)
+		if len(page.JSInfo.AccountDetails) == baseOpts.Limit {
+			offsets[name] = baseOpts.Offset + baseOpts.Limit
+		}
+	}
+
+	// iterate through all the servers that still have pages
+	for len(offsets) > 0 {
+		for name, offset := range offsets {
+			opts := baseOpts
+			opts.EventFilterOptions.Name = name
+			opts.Offset = offset
+
+			page, err := s.JszPing(opts, fopts...)
+			if err != nil {
+				return err
+			}
+			if len(page) == 0 {
+				delete(offsets, name)
+				continue
+			}
+
+			jsz := page[0]
+			fn(jsz.JSInfo.AccountDetails)
+
+			if len(jsz.JSInfo.AccountDetails) == baseOpts.Limit {
+				offsets[name] += baseOpts.Limit
+			} else {
+				delete(offsets, name)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (s *SysClient) CollectClusterAccounts(timeout time.Duration) ([]*server.AccountDetail, error) {
+	accountMap := map[string]*server.AccountDetail{}
+	opts := server.JszEventOptions{
+		JSzOptions: server.JSzOptions{
+			Accounts: true,
+			Streams:  true,
+			Consumer: true,
+			Config:   true,
+		},
+	}
+
+	err := s.FetchJszPaged(opts, 1024, timeout, func(pages []*server.AccountDetail) {
+		for _, acct := range pages {
+			if existing, found := accountMap[acct.Name]; found {
+				existing.Streams = mergeStreams(existing.Streams, acct.Streams)
+			} else {
+				copy := *acct
+				accountMap[acct.Name] = &copy
+			}
+		}
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	var accounts []*server.AccountDetail
+	for _, acct := range accountMap {
+		accounts = append(accounts, acct)
+	}
+
+	sort.Slice(accounts, func(i, j int) bool { return accounts[i].Name < accounts[j].Name })
+	for _, acct := range accounts {
+		sort.Slice(acct.Streams, func(i, j int) bool { return acct.Streams[i].Name < acct.Streams[j].Name })
+	}
+
+	return accounts, nil
+}
+
+func mergeStreams(a, b []server.StreamDetail) []server.StreamDetail {
+	seen := map[string]any{}
+	var result []server.StreamDetail
+
+	for _, s := range append(a, b...) {
+		if _, found := seen[s.Name]; found {
+			continue
+		}
+		seen[s.Name] = struct{}{}
+		result = append(result, s)
+	}
+	return result
+}
+
 func (s *SysClient) Fetch(subject string, data []byte, opts ...FetchOpt) ([]*nats.Msg, error) {
 	if subject == "" {
 		return nil, fmt.Errorf("%w: expected subject 0", ErrValidation)
 	}
 
 	conn := s.nc
-	reqOpts := &FetchOpts{}
+	reqOpts := &FetchOpts{
+		Timeout:     5 * time.Second,
```

Comment:
seems to ignore the supplied timeout, each request should use the supplied I think